services/web-server/test/jwt_test.js

```diff
@@ -0,0 +1,65 @@
+const assert = require('assert');
+const testing = require('taskcluster-lib-testing');
+const taskcluster = require('taskcluster-client');
+const jwt = require('../src/utils/jwt');
+
+suite(testing.suiteName(), () => {
+  const rootUrl = 'https://test.taskcluster.net';
+  // DO NOT USE THESE KEYS IN PRODUCTION
+  const publicKey = [
+    '-----BEGIN PUBLIC KEY-----',
+    'MFswDQYJKoZIhvcNAQEBBQADSgAwRwJActU6+fTN5QIeCPqlyzIFd9zH9FIRG4u8',
+    '3rQm06lcXt/829P07rAx6T6w0t9Pwl7e1YKGIfPxz3ZkaxvnfUalywIDAQAB',
+    '-----END PUBLIC KEY-----',
+  ].join('\n');
+  const privateKey = [
+    '-----BEGIN RSA PRIVATE KEY-----',
+    'MIIBOAIBAAJActU6+fTN5QIeCPqlyzIFd9zH9FIRG4u83rQm06lcXt/829P07rAx',
+    '6T6w0t9Pwl7e1YKGIfPxz3ZkaxvnfUalywIDAQABAkBpuhFW2iJH268zrTLA5wlJ',
+    '1qjzNiwyJHJ8yXmGH8TARFYqm/eINXadE2IiHy6FhiSb+WMFYgNxmKfPF7C/NvKB',
+    'AiEAtNq7m77O98RYFfFiy7ayY2T+KUkgc/s5jX17bPI3MNsCIQCii97tl+NEwcSD',
+    'vl5so+nN9UTP+uYFEZGRO0CHNTU50QIgRbwJlvYt689jf6KBy7b4debdMaImx1HZ',
+    'UTtPOiTzLv0CIF0zB8KBaV0+IPhNCgUOwvtFm4wI1ySm2ylDqQ8dcgRRAiAtON38',
+    'AhSVoX2Pxc5Ym1+IdmBVWaplKg8c5D0TUb1zTg==',
+    '-----END RSA PRIVATE KEY-----',
+  ].join('\n');
+
+  test('should be able to generate a token with a valid expiration', () => {
+    const identity = 'mozilla-auth0/ad|Mozilla-LDAP|haali/';
+    const exp = Math.floor(taskcluster.fromNow('2 days').getTime() / 1000);
+    const { token, expires } = jwt.generate({ rootUrl, privateKey, identity, exp });
+
+    assert.equal(token.length, 352, 'should of had a token');
+    assert.equal(expires.getTime() / 1000, exp, 'should of had a valid expiration');
+  });
+
+  test('should be able to verify a token', async () => {
+    const identity = 'mozilla-auth0/ad|Mozilla-LDAP|haali/';
+    const exp = Math.floor(taskcluster.fromNow('2 days').getTime() / 1000);
+    const { expires, token } = jwt.generate({ rootUrl, privateKey, identity, exp });
+    const decoded = await jwt.verify({ publicKey, token });
+
+    assert.equal(decoded.sub, identity, 'should of had a valid sub property');
+    assert.equal(decoded.iss, rootUrl, 'should of had a valid iss property');
+    assert.equal(decoded.aud, rootUrl, 'should of had a valid aud property');
+    // exp returns a value in seconds; Date.now returns a value in milliseconds
+    // https://github.com/auth0/node-jsonwebtoken#token-expiration-exp-claim
+    assert(decoded.exp * 1000 > Date.now(), 'should of had an expiration in the future');
+    assert.equal(expires.getTime(), decoded.exp * 1000, 'should of had  valid expires value');
+  });
+
+  test('should throw when verifying with the wrong publicKey', async () => {
```

Comment:
It'd be great to add checks for each of the claims being invalid (`aud`, `exp`, etc.)